source/rofi-icon-fetcher.c

```diff
@@ -542,7 +549,15 @@ static void rofi_icon_fetcher_worker(thread_state *sdata,
 
     if (strcmp(entry_name, "") == 0) {
       sentry->query_done = TRUE;
-      rofi_view_reload();
+      g_mutex_lock(&(sentry->widget_list_lock));
+      for (GList *iter = g_list_first(sentry->widget_list); iter;
+           iter = g_list_next(iter)) {
+        rofi_view_reload_widget((widget *)iter->data);
+        widget_unref((widget *)iter->data);
+      }
+      g_list_free(sentry->widget_list);
+      sentry->widget_list = NULL;
+      g_mutex_unlock(&(sentry->widget_list_lock));
```

Comment:
All this part repeats a lot in this diff, it looks like it has to go in a common function